vncviewer/vncviewer.cxx

```diff
@@ -144,20 +162,43 @@ void about_vncviewer()
   fl_message("%s", about_text());
 }
 
-void run_mainloop()
+static void run_mainloop(const char* vncserver, network::Socket* sock)
 {
+  CConn cc(vncServerName, sock);
   int next_timer;
 
-  next_timer = Timer::checkTimeouts();
-  if (next_timer == 0)
-    next_timer = INT_MAX;
+  while (!exitMainloop) {
+    next_timer = Timer::checkTimeouts();
+    if (next_timer == 0)
+      next_timer = INT_MAX;
 
-  if (Fl::wait((double)next_timer / 1000.0) < 0.0) {
-    vlog.error(_("Internal FLTK error. Exiting."));
-    exit(-1);
+    if (Fl::wait((double)next_timer / 1000.0) < 0.0) {
+      vlog.error(_("Internal FLTK error. Exiting."));
+      exit(-1);
+    }
   }
 }
 
+static void mainloop(const char* vncserver, network::Socket* sock)
```

Comment:
`mainloop()` and `run_mainloop()` are a bit too similar. Can we tweak the naming to clarify what the difference is?

Reply:
No good idea.... I tend to merge `run_mainloop` and `mainloop` into one method, as `run_mainloop` is just about 10 lines of code and `mainloop` is also small. Would you agree?

Reply:
Works for me.